ansible_base/authentication/authenticator_plugins/ldap.py

```diff
@@ -410,7 +410,8 @@ def authenticate(self, request, username=None, password=None, **kwargs) -> (obje
             # In unit testing there were cases where the function we are in was being called before get_or_build_user.
             # Its unclear if that was just a byproduct of mocking or a real scenario.
             # Since this call is idempotent we are just going to call it again to ensure the AuthenticatorUser is created for update_user_claims
-            get_or_create_authenticator_user(username, self.database_instance, user_details={}, extra_data=user_from_ldap.ldap_user.attrs.data)
+            user_id = determine_username_from_uid(username, self.database_instance)
```

Comment:
variables are confusing 
- it takes uid and returns username
- there it looks like it takes username and returns int, which is not

Reply:
Yea, it’s kind of weird because it’s taking the uid from ldap and converting it into a username for the Django user table. But the authenticate function traditionally calls the parameter username. I’ll change this up a bit and see what I can come up with.